### DISCUSSION TITLE
Excel outputs

### DISCUSSION THREAD

**Discussion by Nesu195** [firemodels/cfast#2117] [Q&A]
I am using CFAST version 7.7.4. After running my simulation the vents output spreadsheet is excluding the outflow masses. what could it be?

**Comment by preneke** [comment]
I understand your problem. Our problem is that we allow so many vents that the putting all the flows in could easily overwhelm the code or Excel, etc. All we have in the spreadsheet is the net inflow into the first room given for the vent in the vent file. This has been this way for a long time and we have had very little feedback about changing it. If you go to the .out file at each output time there is a table that has a more detailed output of the flow from and to and the layers. It is not complete but it more what your are looking for, I believe.

**Reply by drjfloyd** [reply]
In FDS we have a set of input parameter that define the number of permitted columns for a csv file. If a user has more device or ctrl outputs than the number of columns we simply write multiple outputs. So instead of chid_devc.csv, you get chid_devc_1.csv, chid_devc_2.csv, etc. A similar approach could be taken in CFAST. You could define a set of integer variables for each compartment that denotes which _vent, _target, etc. ouput a particular comparment's data gets written to so that data for a given compartment is contained within one file. For about a decade now Excel has supported over 1000 columns which I think would cover most typical CFAST inputs.

**Comment by rpeacoc** [comment]
I believe that if you run the case with the validation outputs, you get all the vent outputs. If you need even more detail, the _slabs.csv output gives you output for each flow slab in the horizontal flow vents.

**Reply by bjork3** [reply]
Hello, do you have recommendations for how to understand the flow slab method and visualization? I am new to CFAST and I cannot find anything about flow slabs in the Users Guide.
I am in a similar position to @Nesu195 looking for the vent upper layer outflow mass rates. I used the Users_Guide_Example.in file as a starting point. I was expecting to find this data in the _vents.csv file, per the Users Manual:
![_vent csv content](https://github.com/user-attachments/assets/cccfc97e-d52b-489f-8eb4-b64e06283b26)
